source/simulator/melt.cc

```diff
@@ -1689,92 +1692,114 @@ namespace aspect
-                         "if Include melt transport is true. ");
-      prm.declare_entry ("Heat advection by melt", "false",
-                         Patterns::Bool (),
-                         "Whether to use a porosity weighted average of the melt and solid velocity "
-                         "to advect heat in the temperature equation or not. If this is set to true, "
-                         "additional terms are assembled on the left-hand side of the temperature "
-                         "advection equation. Only used if Include melt transport is true. "
-                         "If this is set to false, only the solid velocity is used (as in models "
-                         "without melt migration).");
-      prm.declare_entry ("Use discontinuous compaction pressure", "true",
-                         Patterns::Bool (),
-                         "Whether to use a discontinuous element for the compaction pressure or not. "
-                         "From our preliminary tests, continuous elements seem to work better in models "
-                         "where the porosity is > 0 everywhere in the domain, and discontinuous elements "
-                         "work better in models where in parts of the domain the porosity = 0.");
-      prm.declare_entry ("Average melt velocity", "true",
-                         Patterns::Bool (),
-                         "Whether to cell-wise average the material properties that are used to "
-                         "compute the melt velocity or not. The melt velocity is computed as the "
-                         "sum of the solid velocity and the phase separation flux "
-                         "$ - K_D / \\phi (\\nabla p_f - \\rho_f \\mathbf g)$. "
-                         "If this parameter is set to true, $K_D$ and $\\phi$ will be averaged "
-                         "cell-wise in the computation of the phase separation flux. "
-                         "This is useful because in some models the melt velocity can have spikes "
-                         "close to the interface between regions of melt and no melt, as both $K_D$ "
-                         "and $\\phi$ go to zero for vanishing melt fraction. As the melt velocity is "
-                         "used for computing the time step size, and in models that use heat "
-                         "transport by melt or shear heating of melt, setting this parameter to true "
-                         "can speed up the model and make it mode stable. In computations where "
-                         "accuracy and convergence behavior of the melt velocity is important "
-                         "(like in benchmark cases with an analytical solution), this parameter "
-                         "should probably be set to 'false'.");
+      prm.enter_subsection ("Melt settings");
+      {
+        prm.declare_entry ("Darcy coefficient variation threshold", "1e-3",
```

Comment:
I do not like the `variation` a lot. What about `Relative Darcy coefficient threshold`? Variation sounds like you are measuring the spatial derivatives of darcy coefficients.

Reply:
I feel like it's not really clear what a relative threshold should be, so that name feels a bit weird to me. I thought one other way to express this would be to say, how much is K_D allowed to vary (or deviate) from the reference value? I could call it "Global Darcy coefficient variation threshold" to make clear it doesn't have anything to do with derivatives.

Reply:
Hmm, I am not really happy with that either. Lets discuss it later today.

Reply:
I looked for parameters with a similar meaning in the code, and the closest thing I could find was something like the "Maximum lateral viscosity variation" in the Steinberger material model. But sure, let's discuss this later.